CONTRIBUTING.md

```diff
@@ -1,208 +1,179 @@
+ 8. For IDEA 2017.3 and above, select `Run -> Edit Configurations -> … -> Defaults -> Junit`. Verify that the `Shorten command line setting` is set to `user-local default: none`. You may also need to remove `ant-javafx.jar` from your classpath if that is reported as a source of jar hell.
 
-    Update your local repository with the most recent code from the main Elasticsearch repository, and rebase your branch on top of the latest master branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits.  This makes them easier to review.  As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
+Elasticsearch will now import into your workspace through Gradle. This may take a few minutes.
 
+The Elasticsearch codebase makes heavy use of Java `assert`s and the test runner requires that assertions be enabled within the `JVM`. This can be accomplished by passing the flag `-ea` to the `JVM` on startup. In order to do this in IntelliJ, follow these steps:
 
+ 1. Select `Run -> Edit Configurations -> … -> Defaults -> Junit -> VM Options`
+ 2. Input `-ea`
+
+Wildcard imports (`import java.util.*`) are forbidden and will cause the build to fail. In IntelliJ, follow these steps to tame your `IDE` so it does not make create wildcard imports:
+
+ - Select `Preferences/Settings -> Editor -> Code Style -> Java -> Imports`
+ - There are two configuration options: `Class count to use import with *` and `Names count to use static import with *`. Set their values to `99999` or some other absurdly high value.
+
+**Formatting Guidelines**
+
+ - Java indent is 4 spaces
+ - Line width is 140 characters
+ - The rest is left to Java coding standards
+ - Disable “auto-format on save” to prevent unnecessary format changes. Doing this makes reviews much easier as it gets rid of unnecessary formatting changes. If your IDE supports formatting only modified chunks, that is fine to do.
+ - Do not worry too much about import order. Try not to change it but there is no need to fight your `IDE` to stop it from doing so.
+
+**Preparing to Submit Changes**
+
+ - To create a distribution from the source, follow these steps:
+	1. Using your favorite Command Line Interface (CLI), `cd` into your cloned Elasticsearch folder
+	2. Run `./gradlew assemble`
+ - The package distribution (Debian and RPM) can be found under ./distribution/packages/(deb|rpm)/build/distributions/
+ - The archive distributions (tar and zip) can be found under ./distribution/archives/(tar|zip)/build/distributions/
+ - Before submitting changes, run the test suite to make sure that nothing is broken: `./gradlew check`
+
+**Submitting Changes**
+
+Please follow these steps in order to make sure your code has a chance to make it into the Elasticsearch codebase:
+
+1. Test your changes
```

Comment:
I feel like some of this is duplicated from the above section. I think it might be good enough just to say "after you have tested you change" or something like that.